### PR TITLE
feat: `kafka_oauth` relation interface to mediate Kafka charm and external Auth Servers

### PR DESCRIPTION
**Pull request by marcoppenheimer**
## Motivation
Customers with existing LDAP/AD OAuth flows would like to be able to connect to a Kafka Charm cluster from outside of the Juju model, using OAuth tokens for Authentication

## Proposal
Create an Integrator Charm (similar to the `tls-certificates-operator` charm), that can serve as the intermediary between an existing customer Authorization Server and the Kafka Charm cluster. It will use the `kafka_oauth` relation interface to pass to Kafka the necessary `client-id/secret` and token validation URLs needed to verify incoming connection tokens from external producer/consumer clients.

The Kafka Charm itself will make use of [`strimzi/strimzi-kafka-oauth`](https://github.com/strimzi/strimzi-kafka-oauth) libraries to facilitate principal user generation and ACLs based on client tokens.

## tl;dr
```mermaid
sequenceDiagram
    autonumber
    critical Manual steps by Admin
    🧍->>Auth Servers: Adds Kafka user
    Auth Servers-->>🧍: Kafka client_id/secret
    🧍->>Integrator: Kafka client_id/secret
    end
	loop Within the Juju model
    Kafka Charm->>Integrator: `kafka_oauth` relation
    Integrator-->>Kafka Charm: Kafka client_id/secret + Identity URLs
    end
    Client App->>Kafka Charm: Token
    Kafka Charm->>Auth Servers: Validate token
    Auth Servers-->>Kafka Charm: ✅
    Kafka Charm-->>Client App: ✅
```

## Changes Made
- `feat: add kafka_oauth interface spec`
- `style: switch from 'is expected' terminology to RFC2119 for kafka_client and kafka_oauth`
    - [RFC2119](https://www.rfc-editor.org/rfc/rfc2119) 
    - This can be changed back. More of a suggestion

### CODE REVIEW

**Comment by simskij** [comment]
@marcoppenheimer @massigori is this interface still relevant given https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/oauth/v0/README.md?

**Comment by massigori** [comment]
> @marcoppenheimer @massigori is this interface still relevant given https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/oauth/v0/README.md?

The new `oauth ` interface was built with the idea of allowing support for the client credentials flow required by Kafka. I am not sure if this is used anywhere in the Kafka project but we will not maintain it.
